lightning/src/ln/channelmanager.rs

```diff
@@ -3879,6 +3932,9 @@ where
 												payment_hash,
 												purpose: $purpose,
 												amount_msat,
+												counterparty_skimmed_fee_msat:
+													cmp::max(total_value.saturating_sub(amount_msat), htlcs.iter()
+														.map(|htlc| htlc.counterparty_skimmed_fee_msat.unwrap_or(0)).sum()),
```

Comment:
Why a max here instead of just the sum of the htlc's reported skimmed fees? I'm struggling to think of a case where the htlc sum is less than sender intended total value minus received value

Reply:
Oops yeah, I was looking for a `debug_assert`.